### DISCUSSION TITLE
Jackson and JSONB in the same project

### DISCUSSION THREAD

**Discussion by ctron** [quarkusio/quarkus#29867] [Q&A]
Assuming I have (because legacy) both jackson and JSONB in my quarkus application.

How can I configure a specific reactive REST client to use Jackson over JSONB?

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
There is currently no way to do this.

**Comment by ssaip** [comment]
I also had this problem (which, as it turns out, is actually more of a openapi-generator-maven-plugin problem in my case), so I'm happy to share information if anyone is in the same boat.

We're using jackson right now and I wanted to generate a microprofile client using the openapi-generator-maven-plugin. Unfortunately this can only generate microprofile clients with json-b as of now, so my first approach was to include both jackson and json-b in our quarkus project.

Mixing 2 things that do the same is never a good idea and I wouldn't mind it we keep only jackson - and as far as I understood, [quarkus also prefers jackson over json-b](https://github.com/quarkusio/quarkus/pull/13280) (is this still true?)

Anyway, I managed to workaround this by separately generating models and APIs (thanks to @plevart [comment](https://github.com/OpenAPITools/openapi-generator/issues/8015#issuecomment-1000274752)) and this seems to do the trick.

```
<generatorName>java</generatorName>
<library>native</library>
<generateApis>false</generateApis>
```
and
```
<generatorName>java</generatorName>
<library>microprofile</library>
<generateModels>false</generateModels>
```
